project-addons/stock_custom/models/stock.py

```diff
@@ -79,19 +79,25 @@ def action_done(self):
                         cont += 1
         res = super().action_done()
         for picking in self:
-            if picking.state == 'done' and picking.sale_id and \
-                    picking.picking_type_code == 'outgoing':
-                picking_template = self.env.\
-                    ref('stock_custom.picking_done_template')
-                picking_template.with_context(
-                    lang=picking.partner_id.lang).send_mail(picking.id)
-
             picking_states = picking.sale_id.picking_ids.mapped('state')
             if all(state in ('done', 'cancel') for state in picking_states) \
                     and not all(state == 'cancel' for state in picking_states):
                 picking.sale_id.action_done()
         return res
 
+    @api.multi
+    def write(self, vals):
+        for picking in self:
+            if vals.get('carrier_tracking_ref', False) and \
+                    picking.picking_type_code == 'outgoing' and \
+                    picking.sale_id\
+                    and picking.carrier_tracking_ref is False:
```

Comment:
Creo que esta comprobación no es correcta, sería "not picking.carrier_tracking_ref" ya que no es False, sino None su valor por defecto

Reply:
En las pruebas que he hecho me venía False, pero bueno, de esta forma serviría para ambos casos y me parece más elegante